eo-phi-normalizer/test/eo/phi/rules/new.yaml

```diff
@@ -0,0 +1,180 @@
+title: "Rule set following Nov 2024 revision"
+rules:
+
+  - name: DOT
+    description: 'Accessing an α-binding'
+    pattern: |
+      ⟦ !τ ↦ !b, !B ⟧.!τ
+    result: |
+      ⌈ !b , ⟦ !τ ↦ !b, !B ⟧ ⌉(ρ ↦ ⟦ !τ ↦ !b, !B ⟧)
+    when:
+      - nf: '!b'
+      - nf: '⟦ !B ⟧'
+    tests:
+      - name: 'Contextualization changes ξ'
+        input: ⟦ a ↦ ξ ⟧.a
+        output: ['⟦ a ↦ ξ ⟧ (ρ ↦ ⟦ a ↦ ξ ⟧)']
+      - name: 'Contextualization applies recursively'
+        input: ⟦ a ↦ ξ.b ⟧.a
+        output: ['⟦ a ↦ ξ.b ⟧.b (ρ ↦ ⟦ a ↦ ξ.b ⟧)']
+
+  - name: COPY
+    description: 'Application of α-binding'
+    pattern: |
+      ⟦ !τ1 ↦ ⟦ !τ2 ↦ ∅, !B1 ⟧(!τ2 ↦ !b, !B2), !B3 ⟧
+    result: |
+      ⟦ !τ1 ↦ ⟦ !τ2 ↦ ⌈ !b , ⟦ !τ1 ↦ ⟦ !τ2 ↦ ∅, !B1 ⟧(!τ2 ↦ !b, !B2), !B3 ⟧ ⌉, !B1 ⟧(!B2), !B3 ⟧
+    when:
+      - nf: '!b'
+      - nf: '⟦ !B1 ⟧'
+      - nf: '⟦ !B3 ⟧'
+    tests:
+      - name: 'Should match'
+        input: ⟦ a ↦ ⟦ b ↦ ∅ ⟧ (b ↦ ξ) ⟧
+        output: ['⟦ a ↦ ⟦ b ↦ ⟦ a ↦ ⟦ b ↦ ∅ ⟧ (b ↦ ξ) ⟧ ⟧ () ⟧']
+
+  - name: RHO
+    description: 'Application of a ρ-binding'
+    pattern: |
+      ⟦ !τ ↦ ⟦ !B1 ⟧(ρ ↦ !b, !B2), !B3 ⟧
+    result: |
+      ⟦ !τ ↦ ⟦ !B1, ρ ↦ ⌈ !b , ⟦ !τ ↦ ⟦ !B1 ⟧(ρ ↦ !b, !B2), !B3 ⟧ ⌉ ⟧(!B2), !B3 ⟧
+    when:
+      - nf: '!b'
+      - absent_attrs:
+          attrs: ['ρ']
+          bindings: ['!B1']
+    tests: []
+
+  - name: phi
+    description: 'Accessing a decorated object'
+    pattern: |
+      ⟦!B ⟧.!τ
+    result: |
+      ⟦!B ⟧.φ.!τ
+    when:
+      - nf: '⟦ !B ⟧'
+      - present_attrs:
+          attrs: ['φ']
+          bindings: ['!B']
+      - absent_attrs:
+          attrs: ['!τ']
+          bindings: ['!B']
+    tests: []
+
+  # there's no B2 in the result
+  - name: STAY
+    description: 'Application of a ρ-binding when ρ already exists'
+    pattern: |
+      ⟦ ρ ↦ !b1, !B1 ⟧ (ρ ↦ !b, !B2)
+    result: |
+      ⟦ ρ ↦ !b1, !B1 ⟧
+    when: []
+    tests: []
+
+  - name: OVER
+    description: 'Invalid application (attribute already attached)'
+    pattern: ⟦ !τ ↦ !b1, !B1 ⟧(!τ ↦ !b2, !B2)
+    result: ⊥
+    when:
+      - not_equal: ['!τ', 'ρ']
+    tests:
+      - name: ''
+        input: '⟦ t ↦ ⟦ a ↦ ∅ ⟧ ⟧(t ↦ ⟦ b ↦ ∅ ⟧)'
+        output: ['⊥']
+
+  - name: STOP
+    description: 'Absent attribute access'
+    pattern: |
+      ⟦ !B ⟧.!τ
+    result: |
+      ⊥
+    when:
+      - absent_attrs:
+          attrs: ['!τ', 'φ', 'λ']
+          bindings: ['!B']
+      - nf: '⟦ !B ⟧'
+    tests:
+      - name: 'Accessing nonexistent attribute'
+        input: '⟦ ρ ↦ ⟦ ⟧ ⟧.x'
+        output: ['⊥']
+
+  - name: VOID
```

Comment:
Is it because NULL is some key word in yaml? Bacause in the paper it is named differently. Please rename or leave a comment

Reply:
[e425b67](https://github.com/objectionary/eo-phi-normalizer/pull/523/commits/e425b6778d652e7ac17edacaf53ac74b301defb0)